mypy/types.py

```diff
@@ -447,6 +447,31 @@ def deserialize(cls, data: JsonDict) -> 'TypeVarType':
         )
 
 
+class SelfType(ProperType):
+    def __init__(self, instance: 'Instance', fullname: str, line: int = -1, column: int = -1) -> None:
```

Comment:
We now also require `__slots__` for all classes in `types.py`

Reply:
Will do